okhttp/src/main/kotlin/okhttp3/internal/connection/RealConnection.kt

```diff
@@ -313,7 +316,9 @@ class RealConnection(
     rawSocket?.closeQuietly()
   }
 
-  override fun socket(): Socket = socket!!
+  override fun socket(): Socket =
+    (socket as? RealOkioSslSocket)?.delegate
```

Comment:
I think this hints at being able to unwrap an OkioSocket to an optional platform `Socket?`